### ISSUE TITLE
No information of the Current/Selected page on the selected pagination control is announced by the screen reader.

### ISSUE THREAD

**Issue by msftedad** [radzenhq/radzen-blazor#1259]
**Describe the bug**
No information of the Current/Selected page on the selected pagination control is announced by the screen reader. If the information of the current/selected page is not announced by the narrator, screen reader dependent user will not get the information of the selected page and they may get confused.

**To Reproduce**
1. Open code pen URL- [datagrid pagination (codepen.io)](https://codepen.io/seambr02/pen/KKreLbM).
2. Turn on narrator with 'Ctrl+ Win + Enter' keys.
3. Now navigate to the Pagination controls present in the codepen.
4. Navigate to the Selected link one.
5. Observe the issue.

**Expected behavior**
When we navigate with tab key on the selected Pagination controls for e.g., 1, it should also announce the information that it is the current page, for example link one current page. This can be achieved by defining aria-current property for the controls.

**Screenshots**

https://github.com/radzenhq/radzen-blazor/assets/93735775/ce9e33ed-d9db-4018-ab9c-8d5184d0c1aa

**Desktop (please complete the following information):**
OS Build: Windows 11 Enterprise Insider Preview 22H2 (23541.1000)
Edge Browser: Version 115.0.1880.3 (Official build) dev (64-bit)
Screen Reader: Narrator

**Additional context**
NA

**Comment by enchev** [comment]
Here is what I hear using our latest version

https://github.com/radzenhq/radzen-blazor/assets/5804953/3041a46e-337d-42f8-83ca-823516886f00

**Comment by msftedad** [comment]
Hi team, Any update on this issue?

**Comment by enchev** [comment]
Hey @msftedad,

Have you checked the video I've posted in my previous reply? Do you have any comments?

**Comment by msftedad** [comment]
Hi @enchev, As discussed with the internal team, the current behavior is not good practice.

**Expected Results in Current Scenario:**
Keyboard focus should not move to the complete table when using the Tab key.
Table name must be defined as 'Example'
Keyboard focus should go to the pagination link and the area-current property should be applied to the current/selected page.

**Comment by akorchev** [comment]
> Keyboard focus should not move to the complete table when using the Tab key.

This is a breaking change and we are not willing to make it. 

> Table name must be defined as 'Example'

What is table name?

**Comment by msftedad** [comment]
Hi @akorchev,  As verified and discussed this issue with the team:

**What is the table Name:** When screen reader user navigate the table in Table mode using 'T' shortcut key screen reader should announce the Table Title/Caption (To describe what kind of content present in the table) along with the table row/column information instead of the irrelevant information.

**Current behavior on Pagination controls:** In tab key navigation keyboard focus moves to the pagination control and screen reader focus does not sync with keyboard focus and moves to the pagination region and announce as "Page 1 of 2 (9 items) Go to first page. Go to previous page. Go to page 1. Go to page 2. Go to next page. Go to last page., group" information.
Also, While user navigate the pagination controls using left/right arrow keys screen reader remain silent and doesn't provide the focused control information.

![image](https://github.com/radzenhq/radzen-blazor/assets/93735775/e831504e-2a94-43d9-971d-9b82fb8419ed)


**Expected Behavior:** Scree reader and Keyboard focus should be in sync on pagination controls and screen reader should announce focused control information along with the current state of selection as "Link page 1 current page".

**Comment by akorchev** [comment]
Hi @msftedad,

I am afraid I don't understand much of your post. You are using some tool which I can't find online and we can't verify whether it works correctly or not.

We accept pull requests though. You can open one with the suggested changes and we will gladly review them and merge them if there isn't any breaking behavior. Alternatively you can suggest what changes need to be made - for example set the XXX attribute to YYY when ZZZ happens.
